packages/mui-material/src/Autocomplete/Autocomplete.test.js

```diff
@@ -823,6 +823,99 @@ describe('<Autocomplete />', () => {
       expect(handleSubmit.callCount).to.equal(0);
       expect(handleChange.callCount).to.equal(1);
     });
+
+    it('Should skip disabled options when navigating via keyboard', () => {
+      const { getByRole } = render(
+        <Autocomplete
+          getOptionDisabled={(option) => option === 'two'}
+          openOnFocus
+          options={['one', 'two', 'three']}
+          renderInput={(props) => <TextField {...props} autoFocus />}
+        />,
+      );
+      const textbox = getByRole('combobox');
+
+      act(() => {
+        textbox.focus();
+      });
+
```

Comment:
Because we've set `openOnFocus` for Autocomplete and `autoFocus` for TextField, the autocomplete opens when focused, eliminating the need for this:
```suggestion
```